### ISSUE TITLE
Per-task and dynamically toggle-able diagnostics mode

### ISSUE THREAD

**Issue by tgeoghegan** [divviup/janus#1857]
Generally, we should be conservative about what we log, to avoid adding unactionable noise to logs and to keep log storage costs down. That leads us to only log extraordinary events or events that indicate errors in Janus (as opposed to events that indicate errors in a client or in a peer aggregator).

However, there are cases where we _do_ want a lot more information out of Janus, such as when a new peer aggregator has been paired and we are working out version mismatches or other bugs.

The obvious thing to do would be to modify the Janus deployment to turn up `RUST_LOG` from e.g. `LOG_INFO` to `LOG_DEBUG`. But this is (1) very coarse: it turns up logging for every one of the 100s or 1,000s of tasks Janus is handling whereas we might only care about one or two and (2) even if we could figure out how to craft a `RUST_LOG` line with a suitable `task_id=` value, that would require restarting every Janus component so they can notice the new log settings, and that's too much risk and disruption to the service.

So instead, I propose that we build a per-task diagnostics mode in Janus, and enable operators to toggle it on dynamically. We'd add a column `diagnostics_mode_end` to the `tasks` table. Its value is a timestamp. If NULL, diags mode is off. If set, diags mode is enabled if the current time is before the time. This means operators would able to enable diagnostics for some period while dealing with an incident and would not have to worry about going back and turning it off again.

Then, Janus would check that value before logging events relevant to the task in question, either logging them at `LOG_INFO` instead of `LOG_DEBUG` or logging things it otherwise wouldn't at all.

Then, we'd extend the task representation in the aggregator API so that it takes a time value for `diagnostics_mode_end`. We'd Then wire up something in the `divviup-api` admin console allowing operators to flip on diags mode for tasks.

**Comment by branlwyd** [comment]
SGTM, we should make sure the API is as easy to use as existing logging lines, I suppose taking an additional `task` parameter to allow checking if diagnostic mode is enabled -- something like `diagnostic!(&task, structured_item = ?structured_value, "format string", format_string_values...)`.

N.B. even this, for a shared production service, is probably too much toil (IMO) -- we should aim to make aggregator pairing as easy & self-diagnosable as possible. Of course, that's a high bar -- being able to enable diagnostics for troubleshooting is still a valuable idea as we work towards not needing to troubleshoot.

**Comment by tgeoghegan** [comment]
I agree that orthogonally, we should consider a design for paired aggregator onboarding that doesn't require Divvi Up operator involvement. Something like https://www.ssllabs.com/ssltest/ where you'd provide us a DAP API URL, then we'd automatically run some tests against it and provide a report of which parts don't work. That would work if we have extremely high confidence that there's no bugs in Janus, but what this issue describes helps when we're trying to debug Janus itself.
